app/uk/gov/hmrc/gform/dms/DmsSubmissionController.scala

```diff
@@ -44,36 +50,63 @@ class DmsSubmissionController(
     withJsonBody[DmsHtmlSubmission] { dmsHtmlSubmission =>
       val decodedHtml = decode(dmsHtmlSubmission.html)
       val metadata = dmsHtmlSubmission.metadata
-      val formTemplateId = FormTemplateId(metadata.dmsFormId)
 
-      for {
-        envId <- fileUpload.createEnvelope(formTemplateId, LocalDateTime.now(clock).plusDays(config.formExpiryDays))
-        pdf   <- pdfGenerator.generatePDF(decodedHtml)
-        pdfDoc = documentLoader(pdf)
-        pdfSummary = PdfSummary(pdfDoc.getNumberOfPages.toLong, pdf)
-        _ = pdfDoc.close()
-        submissionRef = SubmissionRef(envId)
-        dmsMetadata = DmsMetaData(formTemplateId, metadata.customerId)
-        submission = Submission(
-          FormId(metadata.dmsFormId),
-          LocalDateTime.now(clock).plusDays(config.formExpiryDays),
-          submissionRef,
-          envId,
-          0,
-          dmsMetadata)
-        // TODO controller should allow data XML submission
-        summaries = PdfAndXmlSummaries(pdfSummary)
-        dmsSubmission = DmsSubmission(
-          metadata.dmsFormId,
-          TextExpression(Constant(metadata.customerId)),
-          metadata.classificationType,
-          metadata.businessArea)
-        _ <- fileUpload.submitEnvelope(submission, summaries, dmsSubmission, 0)
-      } yield {
-        NoContent
+      pdfGenerator.generatePDF(decodedHtml).flatMap { byteArray =>
+        submit(byteArray, metadata)
       }
     }
   }
 
+  def submitPdfToDms: Action[MultipartFormData[Files.TemporaryFile]] = Action.async(parse.multipartFormData) {
```

Comment:
Can you please add some comment explaining why this endpoint exists and who is using it? Sooner or later someone on gform team will ask why this exists when we do not use it.